src/builder/QuarkBuilder.sol

```diff
@@ -123,7 +128,7 @@ contract QuarkBuilder {
             chainAccountsList,
             payment,
             repayIntent.assetSymbol,
-            repayIntent.amount,
+            repayAmount,
```

Comment:
🪄 can you confirm: the value passed to `checkAndInsertWrapOrUnwrapAction` should be our best estimate of the actual repayAmount and not simply `type(uint256).max`, correct?

Reply:
Yes. 
The logics within `checkAndInsertWrapOrUnwrapAction` is essentially trying to wrap/unwrap counter part of meet the input amount. So it will break if it try to wrap/unwrap int256.max of ETH or WETH. Unless we modify max handling in `checkAndInsertWrapOrUnwrapAction` to support it. 

iirc `checkAndInsertWrapOrUnwrapAction` should be put in the last statement insert "right" before the actual action is generated. By then, we should have a number. (such as transferMax, at this stage, it should has user total blanace amount pass into transferAction)